quantling/pyndl
# Adds support for pypoetry (https://python-poetry.org)

This PR introduces the Python packaging and dependency management [poetry](https://python-poetry.org) to this project.

Most stuff is handled in the `pyproject.toml` which is also makes other files (e.g. setup.py, requirements.txt, MANIFEST.in) unnecessary and helps to keep the project cleaner. Further, more and more third-party applications like pytest, pylint, etc are supported to be configured in the toml file. 

To test out, install the package with 
```
poetry install
```

and run the tests with 
```
poetry run pytest
```

This PR is yet, just a draft, as there can (and in my opinion) should be more things cleaned up in the process. This includes:

- [x] check what other configs can be transferred to the pyproject.toml
- [x] Is there any benefit from using tox any longer?
- [x] If we want to keep tox adjust the tox.ini file accordingly, if not adjust the Makefile
- [x] Adapt .travis.yml or setup Github Actions accordingly
